essentials-framework/Essentials Core/PepperDashEssentialsBase/Secrets/CrestronSecretsProvider.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using Crestron.SimplSharp;
+using Crestron.SimplSharp.CrestronDataStore;
+using PepperDash.Core;
+
+
+namespace PepperDash.Essentials.Core
+{
+    public class CrestronSecretsProvider : ISecretProvider
+    {
+        public string Key { get; set; }
+        //Added for reference
+        //private readonly bool _secureSupported;
+        public CrestronSecretsProvider(string key)
+        {
+            Key = key;
+            //Added for future encrypted reference
+            //_secureSupported = CrestronSecureStorage.Supported;
+
+            //if (_secureSupported)
+            //{
+            //      return;
+            //}
+            CrestronDataStoreStatic.InitCrestronDataStore();
+
+        }
+
+        /// <summary>
+        /// Set secret for item in the CrestronSecretsProvider
+        /// </summary>
+        /// <param name="key">Secret Key</param>
+        /// <param name="value">Secret Value</param>
+        public void SetSecret(string key, object value)
+        {
+            var secret = value as string;
+            if (String.IsNullOrEmpty(secret))
+            {
+                Debug.Console(2, this, "Unable to set secret for {0}:{1} - value is empty.", Key, key);

```

Comment:
This should be level 1. It might also make sense to have this method return a bool, with true indicating success. Then, when this method is called from the SecretsManager, it can check for the result and print a message at level 0.